### DISCUSSION TITLE
How does apollo-rs fit into the existing Rust + GraphQL ecosystem?

### DISCUSSION THREAD

**Discussion by danieleades** [apollographql/apollo-rs#779] [Q&A]
hi there! This project looks really cool, i'm glad it exists.

I was hoping you could elaborate on how this crate fits into the existing ecosystem - is this crate competing against, or complimentary to [juniper](https://github.com/graphql-rust/juniper)?

**Comment by lrlna** [comment]
Hi @danieleades, thanks for the question!

We didn't write this project to compete with other projects in the ecosystem. We wanted to have a library with built in semantic analysis to help us write more performant composition and query planning for Apollo Federation implementations along with a few other schema manipulation tools. While it's built to work for some of Apollo's tools, we chose to make it generic enough that other folks can use it as well if they chose to. `juniper` is a really brilliant project, and I think it'd be cool to see if `apollo-parser` could work with it!

**Comment by danieleades** [comment]
i think the strength of Rust is the community. If there's any scope for having apollo and juniper pulling in the same direction, that's got to be a win
